local-cluster/tests/local_cluster.rs

```diff
@@ -3493,146 +3502,200 @@ fn test_fork_choice_refresh_old_votes() {
-            // is a possibility of voting before broadcast has inserted in blockstore.
-            let lighter_fork_latest_vote = wait_for_last_vote_in_tower_to_land_in_ledger(
-                &lighter_fork_ledger_path,
-                &context.lighter_fork_validator_key,
-            )
-            .unwrap();
-            let heaviest_fork_latest_vote = wait_for_last_vote_in_tower_to_land_in_ledger(
-                &heaviest_ledger_path,
-                &context.heaviest_validator_key,
-            )
-            .unwrap();
+            // Wait for blockhashes to expire
+            let mut distance_from_tip: usize;
+            loop {
+                // Get latest votes. We make sure to wait until the vote has landed in
+                // blockstore. This is important because if we were the leader for the block there
+                // is a possibility of voting before broadcast has inserted in blockstore.
+                let lighter_fork_latest_vote = wait_for_last_vote_in_tower_to_land_in_ledger(
+                    &lighter_fork_ledger_path,
+                    &context.lighter_fork_validator_key,
+                )
+                .unwrap();
+                let heaviest_fork_latest_vote = wait_for_last_vote_in_tower_to_land_in_ledger(
+                    &heaviest_ledger_path,
+                    &context.heaviest_validator_key,
+                )
+                .unwrap();
 
-            // Open ledgers
-            let smallest_blockstore = open_blockstore(&smallest_ledger_path);
-            let lighter_fork_blockstore = open_blockstore(&lighter_fork_ledger_path);
-            let heaviest_blockstore = open_blockstore(&heaviest_ledger_path);
-
-            info!("Opened blockstores");
-
-            // Find the first slot on the smaller fork
-            let lighter_ancestors: BTreeSet<Slot> = std::iter::once(lighter_fork_latest_vote)
-                .chain(AncestorIterator::new(
-                    lighter_fork_latest_vote,
-                    &lighter_fork_blockstore,
-                ))
-                .collect();
-            let heavier_ancestors: BTreeSet<Slot> = std::iter::once(heaviest_fork_latest_vote)
-                .chain(AncestorIterator::new(
-                    heaviest_fork_latest_vote,
-                    &heaviest_blockstore,
-                ))
-                .collect();
-            let first_slot_in_lighter_partition = *lighter_ancestors
-                .iter()
-                .zip(heavier_ancestors.iter())
-                .find(|(x, y)| x != y)
-                .unwrap()
-                .0;
+                // Open ledgers
+                {
+                    let smallest_blockstore = open_blockstore(&smallest_ledger_path);
+                    let lighter_fork_blockstore = open_blockstore(&lighter_fork_ledger_path);
+                    let heaviest_blockstore = open_blockstore(&heaviest_ledger_path);
+
+                    info!("Opened blockstores");
```

Comment:
Do we have a log here because open_blockstore can block forever?

Reply:
probably not necessary, removed